### PR TITLE
fix: bad links

### PR DESCRIPTION
**Pull request by nhcarrigan**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.

Closes #XXXXX

CI is currently failing on `main` - this PR fixes the flagged links to resolve that.

### CODE REVIEW

**Comment by erictleung** [comment]
@nhcarrigan thanks for the PR! If you take a look at the recent issues I've created (https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/created_by/erictleung), they correspond to the links you've fixed. So I'm leaving it up to new contributors to slowly fix the CI issues 😄

File: README.md [hunk 1]
```diff
@@ -57,11 +57,11 @@ If you have questions or comments, please create an issue.
 - [The Definitive Guide to Contributing to Open Source](https://www.freecodecamp.org/news/the-definitive-guide-to-contributing-to-open-source-900d5f9f2282/) by [@DoomHammerNG](https://twitter.com/DoomHammerNG)
 - [An Intro to Open Source](https://www.digitalocean.com/community/tutorial_series/an-introduction-to-open-source) - Tutorials by DigitalOcean to guide you on your way to contribution success here on GitHub.
 - [Code Triage](https://www.codetriage.com/) - another, really nice, tool for finding popular repositories and issues filtered by language.
-- [Forge Your Future with Open Source](https://fossforge.com) ($) - Book devoted to explaining open source, how to find a project, and how to start contributing. Inclusive of all roles in software development, not just programmers.
+- [Forge Your Future with Open Source](https://pragprog.com/titles/vbopens/forge-your-future-with-open-source/) ($) - Book devoted to explaining open source, how to find a project, and how to start contributing. Inclusive of all roles in software development, not just programmers.
```
Comment (erictleung) [inline README.md:60]: https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/701

File: README.md [hunk 2]
```diff
@@ -57,11 +57,11 @@ If you have questions or comments, please create an issue.
...
 - [Awesome-for-beginners](https://github.com/MunGell/awesome-for-beginners) - a GitHub repo that amasses projects with good bugs for new contributors, and applies labels to describe them.
 - [Open Source Guides](https://opensource.guide/) - Collection of resources for individuals, communities, and companies who want to learn how to run and contribute to an Open Source project.
 - [45 Github Issues Dos and Don’ts](https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612) - Do's and Don'ts on GitHub.
-- [GitHub Guides](https://guides.github.com/) - basic guides on how to use GitHub effectively.
+- [GitHub Guides](https://docs.github.com/) - basic guides on how to use GitHub effectively.
```
Comment (erictleung) [inline README.md:64]: https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/702

File: README.md [hunk 3]
```diff
@@ -111,21 +111,21 @@ Search links that point directly to suitable issues to contribute to on GitHub.
 
 - [How to choose (and contribute to) your first Open Source project](https://github.com/collections/choosing-projects) by [@GitHub](https://github.com/github)
 - [How to find your first Open Source bug to fix](https://www.freecodecamp.org/news/finding-your-first-open-source-project-or-bug-to-work-on-1712f651e5ba/) by [@Shubheksha](https://github.com/Shubheksha)
-- [First Timers Only](https://kentcdodds.com/blog/first-timers-only/) by [@kentcdodds](https://github.com/kentcdodds)
+- [First Timers Only](https://kentcdodds.com/blog/first-timers-only) by [@kentcdodds](https://github.com/kentcdodds/)
```
Comment (erictleung) [inline README.md:114]: https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/703

File: README.md [hunk 4]
```diff
@@ -111,21 +111,21 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [Getting into Open Source for the First Time](https://www.nearform.com/blog/getting-into-open-source-for-the-first-time/) by [@mcdonnelldean](https://github.com/mcdonnelldean)
 - [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/) by [@GitHub](https://github.com/github)
 - [How to Find a Bug in Your Code](https://8thlight.com/blog/doug-bradbury/2016/06/29/how-to-find-bug-in-your-code.html) by [@dougbradbury](https://twitter.com/dougbradbury)
 - [Mastering Markdown](https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) by [@GitHub](https://github.com/github)
-- [First mission: Contributors page](https://medium.com/@forCrowd/first-mission-contributors-page-df24e6e70705#.2v2g0no29) by [@forCrowd](https://github.com/forCrowd)
+- [First mission: Contributors page](https://forcrowd.medium.com/first-mission-contributors-page-df24e6e70705) by [@forCrowd](https://github.com/forCrowd)
```
Comment (erictleung) [inline README.md:120]: https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/705

File: README.md [hunk 5]
```diff
@@ -111,21 +111,21 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [A Bitter Guide To Open Source](https://medium.com/codezillas/a-bitter-guide-to-open-source-a8e3b6a3c1c4) by [@ken_wheeler](https://medium.com/@ken_wheeler)
 - [A junior developer’s step-by-step guide to contributing to Open Source for the first time](https://hackernoon.com/contributing-to-open-source-the-sharks-are-photoshopped-47e22db1ab86) by [@LetaKeane](https://hackernoon.com/u/letakeane)
 - [Learn Git and GitHub Step By Step (on Windows)](https://medium.com/illumination/path-to-learning-git-and-github-be93518e06dc) by [@ows-ali](https://ows-ali.medium.com/)
 - [Why Open Source and How?](https://careerkarma.com/blog/open-source-projects-for-beginners/) by [@james-gallagher](https://careerkarma.com/blog/author/jamesgallagher/)
 - [How to get started with Open Source - By Sayan Chowdhury](https://www.hackerearth.com/getstarted-opensource/)
-- [What open-source should I contribute to](https://kentcdodds.com/blog/what-open-source-project-should-i-contribute-to/) by [@kentcdodds](https://github.com/kentcdodds)
+- [What open-source should I contribute to](https://kentcdodds.com/blog/what-open-source-project-should-i-contribute-to) by [@kentcdodds](https://github.com/kentcdodds)
```
Comment (erictleung) [inline README.md:128]: https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/706

File: README.md [hunk 6]
```diff
@@ -135,15 +135,15 @@ Search links that point directly to suitable issues to contribute to on GitHub.
 
 ## Using Version Control
 
-- [Think Like (a) Git](http://think-like-a-git.net/) - Git introduction for "advanced beginners", but are still struggling, in order to give you a simple strategy to safely experiment with git.
+- [Think Like (a) Git](https://think-like-a-git.net/) - Git introduction for "advanced beginners", but are still struggling, in order to give you a simple strategy to safely experiment with git.
```
Comment (erictleung) [inline README.md:138]: https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/707

File: README.md [hunk 7]
```diff
@@ -135,15 +135,15 @@ Search links that point directly to suitable issues to contribute to on GitHub.
...
 - [Everyday Git](https://git-scm.com/docs/giteveryday) - A useful minimum set of commands for Everyday Git.
 - [Oh shit, git!](https://ohshitgit.com/) - how to get out of common `git` mistakes described in plain English; also see [Dangit, git!](https://dangitgit.com/) for the page without swears.
 - [Atlassian Git Tutorials](https://www.atlassian.com/git/tutorials) - various tutorials on using `git`.
 - [GitHub Git Cheat Sheet](https://education.github.com/git-cheat-sheet-education.pdf) (PDF)
 - [freeCodeCamp's Wiki on Git Resources](https://forum.freecodecamp.org/t/wiki-git-resources/13136)
 - [GitHub Flow](https://www.youtube.com/watch?v=juLIxo42A_s) (42:06) - GitHub talk on how to make a pull request.
-- [GitHub Learning Resources](https://docs.github.com/en/github/getting-started-with-github/git-and-github-learning-resources) - Git and GitHub learning resources.
+- [GitHub Learning Resources](https://docs.github.com/en/get-started/quickstart/git-and-github-learning-resources) - Git and GitHub learning resources.
```
Comment (erictleung) [inline README.md:146]: https://github.com/freeCodeCamp/how-to-contribute-to-open-source/issues/709

**Comment by nhcarrigan** [comment]
Whoops - thanks @erictleung I should have checked the open issues.

I had a contributor reach out to me in another community inquiring why the CI on their PR was failing, and wanted to get these fixed so that their contribution could be updated.

**Comment by erictleung** [comment]
@nhcarrigan no worries! I know you're not trying to steal any thunder here 😄 I think I did see a community member comment on a PR here, so I'll deal with that accordingly. So I'll close this PR so others can have a hand at contributing. Thanks again for your commitment to helping others! Happy coding!
